app/services/plugin.service.ts

```diff
@@ -145,46 +107,46 @@ export class PluginService {
-        // TODO: consider: this could really just be a cast and 
-        // graph.plugin.context
-        let context: Promise<Context> = this.getPlugin(graph.plugin.kind).script.then((script) => this.getContext(script));
-        return context.then((context: Context): Promise<Program> => {
-            context.sinap.__graph = new Graph(graph);
-            return this.interpretCode
-                .runInContext(context)
-                .then((program) => {
-                    return {
-                        run: (input: ProgramInput): Promise<ProgramOutput> => {
-                            context.sinap.__input = input;
-                            // Cast is necessary. 
-                            // I prefer forcing it to be explicit (and thus not any)
-                            return this.runInputCode.runInContext(context) as Promise<ProgramOutput>;
-                        },
-                        compilationMessages: [""]
-                    };
-                });
-        });
+        if (!(graph.plugin instanceof ConcretePlugin)) {
+            throw "Error: only get interpreters for graphs created with PluginService";
+        }
+
+        let context = this.addToContext(this.getContext(graph.plugin.script), "__graph", new Graph(graph));
+
+        return this.interpretCode
+            .runInContext(context)
+            .then<Program>((program) => {
+                return {
+                    run: (input: ProgramInput): Promise<ProgramOutput> =>
+                        this.runInputCode
+                            .runInContext(this.addToContext(context, "__input", input)),
+                    compilationMessages: [""]
+                };
+            });
     }
 
-    public getPlugin(kind: string): ConcretePlugin {
-        // TODO: Figure out how to cache this without bad performance.
-        const result = this.plugins.get(kind);
-        if (result) {
-            return result;
-        } else {
-            const newPlugin = this.makePlugin(kind);
-            this.plugins.set(kind, newPlugin);
-            return newPlugin;
+    public getPlugin(kind: string): Promise<ConcretePlugin> {
+        const loadedPlugin = this.plugins.get(kind);
+        if (loadedPlugin) {
+            return loadedPlugin;
         }
+        const plugin = this.makePlugin(kind);
+        this.plugins.set(kind, plugin);
+        return plugin;
     }
 
-    private makePlugin(kind: string): ConcretePlugin {
+    public makePlugin(kind: string): Promise<ConcretePlugin> {
         switch (kind) {
             case "dfa.sinap.graph-kind":
-                const plugin = new ConcretePlugin();
-                plugin.script = this.fileService.readFile("./build/plugins/dfa-interpreter.js")// TODO: Put real file in here.
-                    .then((code: string) => this.sandboxService.compileScript(code));
-                return plugin;
+                let defintions = this.fileService.readFile("./dfa-definition.sinapdef")
```

Comment:
With Webpack you could bring it into the bundle with something like, `.readFile(require("file!./../dfa-definition.sinapdef")`